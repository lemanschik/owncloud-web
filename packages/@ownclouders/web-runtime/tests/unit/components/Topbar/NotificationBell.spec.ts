import NotificationBell from 'web-runtime/src/components/Topbar/NotificationBell.vue'
import {
  createStore,
  defaultComponentMocks,
  defaultPlugins,
  defaultStoreMockOptions,
  mount
} from 'web-test-helpers/src'

describe('NotificationBell', () => {
  it('should match snapshot', async () => {
    const { wrapper } = getWrapper()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('should not render badge if 0 notifications', async () => {
    const { wrapper } = getWrapper({})
    await wrapper.setProps({ notificationCount: 0 })
    expect(wrapper.find('.badge').exists()).toBeFalsy()
  })
  it.each([
    [1, '1'],
    [11, '11'],
    [99, '99'],
    [110, '99+']
  ])("should render badge count '%s'", async (input, expected) => {
    const { wrapper } = getWrapper({})
    await wrapper.setProps({ notificationCount: input })
    expect(wrapper.find('.badge').text()).toBe(expected)
  })
  it('displays a tooltip with the notifications label', () => {
    const { wrapper } = getWrapper({
      mountType: mount,
      props: {
        notificationCount: 0
      }
    })
    expect(wrapper.find('#oc-notifications-bell').attributes('aria-label')).toEqual('Notifications')
  })
  it('animates when notification count changes', async () => {
    const { wrapper } = getWrapper({
      props: {
        notificationCount: 5
      }
    })
    wrapper.setProps({ notificationCount: 10 })
    await wrapper.vm.$nextTick()
    expect(wrapper.find('.shake').exists()).toBe(true)
  })
})
function getWrapper({ mountType = mount, mocks = {}, props = {} } = {}) {
  const localMocks = { ...defaultComponentMocks(), ...mocks }
  const storeOptions = {
    ...defaultStoreMockOptions
  }
  const store = createStore(storeOptions)
  return {
    mocks: localMocks,
    storeOptions,
    wrapper: mountType(NotificationBell, {
      global: {
        renderStubDefaultSlot: true,
        plugins: [...defaultPlugins(), store],
        mocks: localMocks,
        stubs: { 'oc-icon': true }
      }
    })
  }
}
