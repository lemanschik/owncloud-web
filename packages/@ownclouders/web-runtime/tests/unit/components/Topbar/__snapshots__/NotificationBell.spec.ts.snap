// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationBell should match snapshot 1`] = `
<button aria-label="Notifications" class="oc-button oc-rounded oc-button-m oc-button-justify-content-center oc-button-gap-m oc-button-brand oc-button-brand-raw-inverse" id="oc-notifications-bell" type="button">
  <!-- @slot Content of the button -->
  <oc-icon-stub class="oc-cursor-pointer oc-flex oc-flex-middle" fill-type="line" name="notification-3"></oc-icon-stub>
  <!--v-if-->
</button>
`;
